persist/fs/read.go

```diff
@@ -209,28 +345,34 @@ func (r *reader) readInfo(size int) error {
 	r.blockSize = time.Duration(info.BlockSize)
 	r.entries = int(info.Entries)
 	r.entriesRead = 0
+	r.metadataRead = 0
 	return nil
 }
 
-func (r *reader) readIndex(size int) error {
-	// NB(r): need to entirely consume index file so we don't stripe
-	// between index and data contents as we read the data contents.
-	r.prepareUnreadBuf(size)
-	n, err := r.indexFdWithDigest.ReadAllAndValidate(r.unreadBuf[:size], r.expectedIndexDigest)
-	if err != nil {
-		return err
+func (r *reader) readIndex() error {
+	r.decoder.Reset(r.indexDecoderStream)
+	for i := 0; i < r.entries; i++ {
```

Comment:
Looks like we're still keeping all of the index file in memory, is the plan to do this lazily in subsequent PRs?

Reply:
Correct, however we'll implement this in the seeker and keep the reader as is.  We'll just never use the reader outside needing to do batch jobs.